_posts/2024-03-22-kraft-migration.md

```diff
@@ -0,0 +1,388 @@
+
+While the migration is going on, you can see the cluster changing the metadata state during the migration process from `ZooKeeper` to `KRaftPostMigration`, and then waiting for the user to finalize it.
+
+```shell
+NAME         DESIRED KAFKA REPLICAS   DESIRED ZK REPLICAS   READY   METADATA STATE   WARNINGS
+my-cluster   3                        3                     True    ZooKeeper        
+my-cluster   3                        3                     True    ZooKeeper        
+my-cluster   3                        3                     True    ZooKeeper        True
+my-cluster   3                        3                             ZooKeeper        True
+my-cluster   3                        3                             ZooKeeper        True
+my-cluster   3                        3                             ZooKeeper        
+my-cluster   3                        3                             ZooKeeper        
+my-cluster   3                        3                             ZooKeeper        
+my-cluster   3                        3                     True    KRaftMigration   
+my-cluster   3                        3                     True    KRaftDualWriting   
+my-cluster   3                        3                     True    KRaftPostMigration   
+my-cluster   3                        3                     True    KRaftPostMigration   True
+```
+
+In this status, the cluster is still working in "dual-write" mode and the user can validate that everything is working fine before deciding to finalize the migration or rolling back to use ZooKeeper.
+
+#### Migration finalization
+
+At this point, the user can finalize the migration by changing the `strimzi.io/kraft` annotation to `enabled`.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+  annotations:
+    strimzi.io/kraft: enabled # updated from "migration" to "enabled" to finalize the migration process
+    strimzi.io/node-pools: enabled
+# ...
+```
+
+The operator will configure the KRaft controllers without the ZooKeeper connection, with the migration flag disabled and roll them.
+When the cluster is full KRaft, the operator will delete all the resources related to ZooKeeper as well.
+
+During the finalization, the cluster metadata state moves to the final `KRaft`.
+
+NAME         DESIRED KAFKA REPLICAS   DESIRED ZK REPLICAS   READY   METADATA STATE   WARNINGS
+...
+my-cluster   3                        3                     True    KRaftPostMigration   True
+my-cluster   3                        3                     True    PreKRaft             True
+my-cluster   3                        3                     True    KRaft                True
+my-cluster                                                  True    KRaft                True
+my-cluster                                                  True    KRaft                
```

Comment:
```suggestion
```shell
NAME         DESIRED KAFKA REPLICAS   DESIRED ZK REPLICAS   READY   METADATA STATE   WARNINGS
...
my-cluster   3                        3                     True    KRaftPostMigration   True
my-cluster   3                        3                     True    PreKRaft             True
my-cluster   3                        3                     True    KRaft                True
my-cluster                                                  True    KRaft                True
my-cluster                                                  True    KRaft                
```
```

Reply:
I don't get what's the suggestion here.

Reply:
the ```shell wrapping

Reply:
why removing it? It's wrapping all the other console outputs I have across the blog post.

Reply:
I was actually suggesting to add them, because they were missing :)

Reply:
LOL!!! Thanks Fede!